File: fe-ta/src/models/HeroSection.ts
import { de } from "element-plus/es/locales.mjs";
import type { HeroSectionProductResponse } from "./HeroSectionProduct";

export interface HeroSectionRequest {
  title?: string;
  description?: string;
  pageHero?: string;
  heroMediaFile?: File | null;
  isPublished?: boolean;
  publishFrom?: string;
  publishTo?: string;
  createUid?: number;
  writeIUid?: number;
  updateTime?: string;
  note?: string;
  option1?: string;
  option2?: string;
  option3?: string;
  option4?: string;
  option5?: string;
}

export interface HeroSectionFilterRequest {
  title?: string;
  description?: string;
  pageHero?: string;
  isPublished?: boolean;
  publishFrom?: string;
  publishTo?: string;
}

export interface HeroSectionResponse {
  id: number;
  title?: string;
  description?: string;
  heroMediaUrl?: string;
  pageHero?: string;
  heroMediaType?: string;
  isPublished: boolean;
  publishFrom?: string; // ISO date string
  publishTo?: string; // ISO date string
  createUid?: number;
  writeIUid?: number;
  updateTime?: string; // ISO date string
  note?: string;
  option1?: string;
  option2?: string;
  option3?: string;
  option4?: string;
  option5?: string;

  heroProducts?: HeroSectionProductResponse[];
}

File: beSQLSugar/beSQLSugar/Infrastructure/Database/Enities/HeroSection.cs
using SqlSugar;
namespace beSQLSugar.Infrastructure.Database.Enities
{
    [SugarTable("hero_section")]
    public class HeroSection
    {
        [SugarColumn(IsPrimaryKey = true, IsIdentity = true)]
        public int Id { get; set; }

        [SugarColumn(ColumnDataType = "NVARCHAR(256)")]
        public string? Title { get; set; }

        [SugarColumn(ColumnDataType = "NVARCHAR(256)")]
        public string? Description { get; set; }

        [SugarColumn(ColumnDataType = "NVARCHAR(256)")]
        public string? PageHero { get; set; }

        [SugarColumn(ColumnDataType = "NVARCHAR(256)")]
        public string? HeroMediaUrl { get; set; }

        [SugarColumn(ColumnDataType = "NVARCHAR(20)")]
        public string? HeroMediaType { get; set; }

        [SugarColumn(DefaultValue = "0")]
        public bool IsPublished { get; set; }

        [SugarColumn(IsNullable = true)]
        public DateTime? PublishFrom { get; set; }

        [SugarColumn(IsNullable = true)]
        public DateTime? PublishTo { get; set; }

        [SugarColumn(IsNullable = true)]
        public int? CreateUid { get; set; }

        [SugarColumn(ColumnDataType = "NVARCHAR(100)", IsNullable = true)]
        public string? CreatedName { get; set; }

        [SugarColumn(IsNullable = true)]
        public int? WriteIUid { get; set; }
        
        [SugarColumn(ColumnDataType = "NVARCHAR(100)", IsNullable = true)]
        public string? UpdatedName { get; set; }

        [SugarColumn(IsNullable = true)]
        public DateTime? UpdateTime { get; set; }

        [SugarColumn(ColumnDataType = "NVARCHAR(250)", IsNullable = true)]
        public string? Note { get; set; }

        [SugarColumn(Length = 120, IsNullable = true)]
        public string? Option1 { get; set; }

        [SugarColumn(Length = 120, IsNullable = true)]
        public string? Option2 { get; set; }

        [SugarColumn(Length = 120, IsNullable = true)]
        public string? Option3 { get; set; }

        [SugarColumn(Length = 120, IsNullable = true)]
        public string? Option4 { get; set; }

        [SugarColumn(Length = 120, IsNullable = true)]
        public string? Option5 { get; set; }

        // Quan hệ One-to-Many: HeroSection -> HeroSectionProducts
        [Navigate(NavigateType.OneToMany, nameof(HeroSectionProduct.HeroSectionId))]
        public List<HeroSectionProduct>? HeroProducts { get; set; } 
    }
}
